### PR TITLE
Lecture 4: Exact inference

### PR DESCRIPTION
**Pull request by ks-korovina**
Hi Xun,

Here are our notes for Lecture 4 (also at https://ks-korovina.github.io/pgm-spring-2019/notes/lecture-04/). We are looking forward to your review. Thank you!

Ksenia, Austin, Lingxiao, Tingfung

### CODE REVIEW

File: _posts/2019-01-28-lecture-04.md [hunk 1]
```diff
@@ -0,0 +1,681 @@
...
+
+## Introduction
+In this previous lectures, we introduce the concept of **Graphical Models** and its mathematical formulations. 
+Now we know that we can use a graphical model $M$ (Bayesian network or undirected graph model) to specify a probability distribution $P_{M}$ satisfying some conditional independence property. 
+In this lecture, we will study how to utilize a graphical model. Given a GM $M$, we generally have two type of tasks 
++ **Inference:** answering queries about the probability distribution $P_M$ defined by $M$, for examples, <script type="math/tex">P_M(X|Y)</script> where $X$ and $Y$ are subsets of variables in GM $M$.
++ **Learning:** estimating a plausible model $M$ from data $D$.  We call the process of obtaining a point estimate of $M$ as *learning*, but for Bayesian, they seek the posterior distribution of <script type="math/tex">p(M|D)</script>, which is actually an *inference* problem. The learning is task is highly related to the inference task. When we want to compute a point estimate of $M$, we need to do inference to impute the missing data if not all the variables are observable. So the learning algorithm usually uses inference as a subroutine.
```
Comment (xunzheng) [inline _posts/2019-01-28-lecture-04.md]:
```suggestion
+ **Learning:** estimating a plausible model $M$ from data $D$.  We call the process of obtaining a point estimate of $M$ as *learning*, but for Bayesian, they seek the posterior distribution of <script type="math/tex">p(M|D)</script>, which is actually an *inference* problem. The learning task is highly related to the inference task. When we want to compute a point estimate of $M$, we need to do inference to impute the missing data if not all the variables are observable. So the learning algorithm usually uses inference as a subroutine. 
```

File: _posts/2019-01-28-lecture-04.md [hunk 2]
```diff
@@ -0,0 +1,681 @@
...
+    </div>
+  </div>
+  <figcaption>
+    <strong>A Deep Belief Network with 3 hidden layers for image processing. </strong> 
+  </figcaption>
+</figure>
+
+#### Most Probable Assignment
+Another interesting query is to find the **most probable joint assignment**(MPA) for *some* variables in interest. 
+Such reasoning is usually performed under some evidence $\mathbf{e}$ and ignoring some "don'care" variables $Z$,
```
Comment (xunzheng) [inline _posts/2019-01-28-lecture-04.md]:
```suggestion
Such reasoning is usually performed under some evidence $\mathbf{e}$ and ignoring some "don't care" variables $Z$,
```

File: _posts/2019-01-28-lecture-04.md (hunk 1 above)
Comment (xunzheng) [inline _posts/2019-01-28-lecture-04.md]:
```suggestion
+ **Learning:** estimating a plausible model $M$ from data $D$.  We call the process of obtaining a point estimate of $M$ as *learning*, but for Bayesian, they seek the posterior distribution of <script type="math/tex">p(M|D)</script>, which is actually an *inference* problem. The learning task is highly related to the inference task. When we want to compute a point estimate of $M$, we need to do inference to impute the missing data if not all the variables are observable. So the learning algorithm usually uses inference as a subroutine. 
```

File: _posts/2019-01-28-lecture-04.md [hunk 3]
```diff
@@ -0,0 +1,681 @@
...
+
+### Complexity Analysis in Graph Perspective
+
+In the beginning of this section we have argued that the bottleneck of VE’s complexity is determined by the scope size of the maximum intermediate factor generated in the procedure of VE. In above subsection we have shown that each intermediate factor in VE is an elimination clique in graph elimination algorithm, and the largest elimination clique is also a largest maximal clique in reconstituted graph.
+
+Then, given an elimination ordering $I$, the problem of get the bottleneck maximum scope size $N_{max}$ of intermediate factors is equivalent of finding the largest clique in reconstituted graph $G’_I(V, E’)$, which is a pure graph theoretic question. And there are efficient mature algorithm for solving this problem. 
+
+### Elimination Ordering 
+
+We can define the **width** of a reconstituted graph as the size of largest clique minus 1. Let $w_{G,I}$ be the width of $G’_I(V, E’)$. For different ordering $I$, $w_{G,I}$ is different.
```
Comment (xunzheng) [inline _posts/2019-01-28-lecture-04.md]:
```suggestion
We can define the **width** of a reconstituted graph as the size of largest clique minus 1. Let $w_{G,I}$ be the width of $G_I'(V, E’)$. For different ordering $I$, $w_{G,I}$ is different. 
```

Comment (xunzheng) [reply _posts/2019-01-28-lecture-04.md]: Not sure why is this breaking up in the website..

**Review by xunzheng — CHANGES_REQUESTED** [review]

File: _posts/2019-01-28-lecture-04.md (hunk 3 above)
Comment (ks-korovina) [reply _posts/2019-01-28-lecture-04.md]: We couldn't figure that out either, hopefully your suggestion helps.

**Review by xunzheng — APPROVED** [review]

**Pull request merged**